Docs/source/theory/pic.rst

```diff
@@ -434,6 +434,85 @@ of this model can be found in the section
+
+   Diagrams illustrating various time dependencies of the current density :math:`\mathbf{\tilde{J}}` and charge density :math:`\tilde{\rho}` for constant/linear (CL), both constant (CC), linear (LL) and quadratic (QQ) dependencies with :math:`m` subintervals: (first column) :math:`m=1`, (second) :math:`m=2` and (third) :math:`m=4`. CL1 corresponds to the standard PSATD PIC method. The triangle and circle glyphs represent the times at which the macroparticles deposit :math:`\mathbf{\tilde{J}}` and :math:`\tilde{\rho}` on the grid, respectively. The dashed and solid lines represent the assumed time dependency of :math:`\mathbf{\tilde{J}}` and :math:`\tilde{\rho}` within one timestep, when integrating the Maxwell equations analytically.
+
+Using the piecewise definition of :math:`\hat{\rho}` and :math:`\mathbf{\tilde{J}}`, the modified Maxwell's equations can be integrated analytically over one timestep :math:`\Delta t`, i.e., from :math:`t=n\Delta t` to :math:`t=(n+1)\Delta t`. In practice, this is done by sequentially integrating these equations over each subinterval :math:`\ell \in [0,m-1]`.
+Final discretized equations write as:
+
+.. math::
+
+   \begin{align}
+   \begin{split}
+   {\mathbf{\tilde{E}}}^{n+(\ell+1)/m} & = C{\mathbf{\tilde{J}}}^{n+\ell/m}+ic^2\frac{S}{ck}\mathbf{k}\times{\mathbf{\tilde{J}}}^{n+\ell/m}+ic^2\frac{S}{ck}\tilde{F}^{n+\ell/m}\mathbf{k} \\
+   &+ \frac{1}{\varepsilon_0 ck}\left(Y_3\mathbf{a_J} + Y_2\mathbf{b_J} - S\mathbf{c_J}\right)
+   + \frac{ic^2}{\varepsilon_0 c^2k^2}\left({Y_1}a_{\rho}-Y_{5}b_{\rho}-Y_{4}c_{\rho}\right)\mathbf{k},
+   \\[4pt]
+   \end{split}
+   \\[4pt]
+   \begin{split}
+   {\mathbf{\tilde{B}}}^{n+(\ell+1)/m} & = C {\mathbf{\tilde{B}}}^{n+\ell/m}-i\frac{S}{ck}\mathbf{k}\times{\mathbf{\tilde{E}}}^{n+\ell/m}
+   - \frac{i}{\varepsilon_0 c^2k^2}\mathbf{k}\times\left(Y_1\mathbf{a_J} -Y_5\mathbf{b_J} -Y_4\mathbf{c_J} \right),
+   \end{split}
+   \\[4pt]
+   \begin{split}
+      {{\tilde{F}}}^{n+(\ell+1)/m} & = C \tilde{F}^{n+\ell/m}+i\frac{S}{ck}\mathbf{k} \cdot {\mathbf{\tilde{E}}}^{n+\ell/m}+\frac{i}{\varepsilon_0 c^2k^2}\mathbf{k}\cdot\left(Y_1\mathbf{a_J}-Y_5\mathbf{b_J}-Y_4\mathbf{c_J}\right) \\
+   &+ \frac{1}{\varepsilon_0 ck}\left({Y_3}a_{\rho}+{Y_2}b_{\rho}-Sc_{\rho}\right),
+   \end{split}
+   \end{align}
+
+where
+
+.. math::
+
+   \begin{aligned}
+   C &= \cos(ck\delta t), \ S = \sin(ck\delta t),
+   \\
+   Y_1 & = \frac{(1-C)(8-c^2k^2\delta t^2)-4Sck\delta t}{2 c^2 k^2 \delta t^2},
+   \\
+   Y_2 & = \frac{2(C-1)+ S ck\delta t }{2 ck\delta t},
+   \\
+   Y_3 & = \frac{S(8- c^2k^2\delta t^2 ) - 4ck\delta t(1+C)}{2c^2 k^2 \delta t^2},
+   \\
+   Y_4 &= (1-C), \ Y_5 = \frac{(1+C) ck\delta t - 2S}{2ck \delta t}.
+   \end{aligned}
+
+Here, :math:`\mathbf{a_J}, \mathbf{b_J}, \mathbf{c_J}, a_{\rho}, b_{\rho}, c_{\rho}` are polynomial coefficients based on the time dependencies of the current  and charge densities, given in Table I:
+
+.. _fig-j_rho_table:
+
+.. figure:: j_rho_table.png
```

Comment:
Can we store this on gist.github.com and add the raw link here?

Reply:
@oshapoval
Would you document the few steps you followed to add the image with gist.github.com and add it somewhere in the docs page linked in the message above? I think we can easily include it in this PR.